src/Features/Core/Portable/Completion/CompletionServiceWithProviders.cs

```diff
@@ -194,6 +218,12 @@ internal protected CompletionProvider GetProvider(CompletionItem item)
             {
                 lock (_gate)
                 {
+                    if (!_roleProvidersInitialized)
+                    {
+                        CreateRoleProviders(ImmutableHashSet.Create<string>());
+                        _roleProvidersInitialized = true;
+                    }
+
                     _nameToProvider.TryGetValue(name, out provider);
```

Comment:
❓ Why not change this to a `GetOrAdd` call that initializes the requested provider on demand, similar to how `_rolesToProviders` is used above? It seems this would be a clean (non-hacky) way to address the problem.